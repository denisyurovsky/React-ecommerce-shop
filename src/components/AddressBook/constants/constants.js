export const LISTBOX_PADDING = 8;
export const DOUBLE_LISTBOX_PADDING = LISTBOX_PADDING * 2;
export const LIMIT_NUMBER_ITEMS = 8;
export const ITEM_SIZE_LG = 48;
export const ITEM_SIZE_MD = 36;
