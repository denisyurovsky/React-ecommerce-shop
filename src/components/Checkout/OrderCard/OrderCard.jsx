import { Card, CardMedia, Typography } from '@mui/material';
import { Box } from '@mui/system';
import PropTypes from 'prop-types';
import React from 'react';

import noImg from '../../../assets/images/noImg.png';

import styles from './OrderCard.module.scss';

const OrderCard = ({ images, name, price, discountPrice, quantity }) => {
  return (
    <Card data-cy="orderCard" className={styles.card}>
      <CardMedia
        component="img"
        image={images.length ? images[0] : noImg}
        alt={name}
        className={styles.image}
      />
      <Typography gutterBottom className={styles.name}>
        {name}
      </Typography>
      <Box className={styles.priceContainer}>
        <Typography>qty: {quantity}</Typography>
        <Typography>{discountPrice && <s>{price} $</s>}</Typography>
        <Typography>{discountPrice ? discountPrice : price} $</Typography>
      </Box>
    </Card>
  );
};

OrderCard.propTypes = {
  images: PropTypes.array.isRequired,
  name: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  quantity: PropTypes.number,
  discountPrice: PropTypes.number,
};

export default OrderCard;
