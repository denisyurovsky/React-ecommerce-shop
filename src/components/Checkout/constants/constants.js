export const PANEL = {
  DELIVERY_ADDRESS: 'Delivery Address',
  PERSONAL_INFORMATION: 'Personal Information',
  PERSONAL_ADDRESS: 'Personal Address',
  PAYMENT_METHOD: 'Payment method',
  ORDER_CONFIRMATION: 'Order confirmation',
};
