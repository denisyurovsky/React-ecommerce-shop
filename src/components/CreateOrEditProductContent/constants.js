export const PRICE_ERROR = 'Only numbers';
export const DESCRIPTION_ERROR = 'Only latin characters';
export const LENGTH_ERROR = 'Should contain at least one symbol';
export const DISCOUNT_ERROR = 'Discount should be less than price';
export const SELECTION_ERROR = 'Choose category';
