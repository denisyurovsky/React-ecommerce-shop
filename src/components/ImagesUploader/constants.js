export const LOSS_OF_IMAGES = 'We lost images';

export const STATUS = {
  CROPPED: 'cropped',
  CROPPING: 'cropping',
  IDLE: 'idle',
};

export const TEST_IMG_URL = 'http://someURL';
