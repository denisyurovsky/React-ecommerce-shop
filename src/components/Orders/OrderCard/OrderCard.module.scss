@import "../../../assets/styles/colors.scss";
@import "../../../assets/styles/breakpoints.scss";

.collapseButton {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: flex-start !important;
  height: 70px;
  padding-left: 20px !important;
  padding-right: 20px !important;
  padding-top: 14px !important;
  background-color: $grey100 !important;

  @media all and (max-width: $sm) {
    padding-left: 14px !important;
    padding-top: 18px !important;
  }
}

.collapseInner {
  padding-left: 4px;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;

  @media all and (max-width: $sm) {
    padding: 0 !important;
    flex-direction: column;
    height: fit-content;
  }
}

@media all and (max-width: $sm) {
  .cardContent {
    padding-left: 14px !important;
    padding-top: 10px !important;
  }
}

.dateInformation {
  display: flex;
  flex-direction: column;
}

.priceInformation {
  display: flex;
  flex-direction: row;
  align-items: flex-start !important;
}

.card {
  margin-bottom: 30px;
  width: 100%;
}

.image {
  width: 80px !important;
  margin-right: 20px;
  height: 100%;
}

.orderDate {
  font-weight: bold !important;

  @media all and (max-width: $md) {
    font-size: 14px !important;
  }
}

.orderPrice {
  margin-right: 14px !important;
  margin-top: 8px !important;
  font-size: 22px !important;
  font-weight: 900 !important;

  @media all and (max-width: 450px) {
    margin: 0 !important;
    margin-top: 10px !important;
    font-size: 14px !important;
  }
}

.showMoreButton {
  margin-top: 10px;

  :hover {
    cursor: pointer;
  }
}

.orderPaymentStatus {
  font-size: 12px !important;
  padding-top: 4px;
  margin-right: 30px !important;
  vertical-align: bottom;

  @media all and (max-width: 450px) {
    font-size: 10px !important;
    margin-right: 0 !important;
  }
}

.actionButton {
  text-transform: uppercase !important;
  box-sizing: border-box;
  font-size: 14px !important;
  height: 38px !important;
  padding: 0;
  padding-left: 15px;
  padding-right: 15px;
  margin-top: 10px !important;

  @media all and (max-width: $sm) {
    display: none !important;
  }
}

.mobileActionButton {
  width: 90%;
  display: none !important;
  margin-bottom: 10px !important;

  @media all and (max-width: $sm) {
    display: unset !important;
  }
}

.deliveryDestination {
  font-weight: bold !important;

  @media all and (max-width: 450px) {
    font-size: 14px !important;
  }
}

.deliveryInformation {
  font-weight: bold !important;
  position: relative;
  display: flex;
  flex-direction: column;
}

.deliveryValue {
  font-weight: normal !important;
}

.outerInformation {
  display: flex;
}

.deliveryMark {
  padding-left: 14px !important;
  padding-right: 14px !important;
  padding-top: 4px !important;
  font-size: 12px !important;
  vertical-align: middle;
  height: 26px;
  margin-left: 12px !important;
  border-radius: 20px;
  color: $greyLight;

  @media all and (max-width: $md) {
    margin: 0 !important;
    margin-left: 2px !important;
    margin-top: 4px !important;
    height: unset !important;
    padding: 7px !important;
    border-radius: 50% !important;
  }
}

.deliveryMark1 {
  padding-right: 20px;
  padding-left: 20px;
  background-color: $green500 !important;
}

.deliveryMark2 {
  padding-right: 20px;
  padding-left: 20px;
  background-color: $green500 !important;
}

.deliveryMark3 {
  background-color: $greyA400 !important;
}

.deliveryMark4 {
  background-color: $greyDark !important;
}

.sliderContainer {
  margin-top: 15px !important;
  margin-bottom: 15px !important;
  width: 300px !important;
  height: 100px !important;
  height: fit-content;
  margin-right: 60px;
  position: relative;

  path {
    color: black;
  }

  img {
    width: 80px !important;
    height: 90px;

    @media all and (max-width: $sm) {
      max-height: 350px !important;
      object-fit: contain !important;
      width: 100% !important;
      height: 100% !important;
    }
  }

  @media all and (max-width: $md) {
    width: 200px !important;
  }

  @media all and (max-width: $sm) {
    width: 80% !important;
    margin: 0 !important;
    height: auto !important;
    margin-top: 15px !important;

    svg {
      top: 150px !important;
    }
  }
}
