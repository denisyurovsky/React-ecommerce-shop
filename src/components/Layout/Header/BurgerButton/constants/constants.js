import { pathNames } from '../../../../../constants/pathNames';

export const LINKS = [
  {
    name: 'HOME',
    link: '/',
  },
  {
    name: 'PRODUCTS',
    link: pathNames.PRODUCTS,
  },
];
