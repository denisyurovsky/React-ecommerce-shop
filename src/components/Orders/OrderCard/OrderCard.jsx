import { ExpandMore } from '@mui/icons-material';
import ArrowBackIosNewIcon from '@mui/icons-material/ArrowBackIosNew';
import ArrowForwardIosIcon from '@mui/icons-material/ArrowForwardIos';
import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
import {
  Box,
  Card,
  CardActions,
  CardContent,
  CardMedia,
  Typography,
} from '@mui/material';
import Collapse from '@mui/material/Collapse';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import React, { useEffect, useState } from 'react';
import { useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import Slider from 'react-slick';
import { toast } from 'react-toastify';

import productsApi from '../../../api/products';
import noImg from '../../../assets/images/noImg.png';
import { notificationError, BREAK_POINT } from '../../../constants/constants';
import { orderStatus, orderState } from '../../../constants/orderStatus';
import { pathNames } from '../../../constants/pathNames';
import {
  formatDateWithFullMonth,
  formatPrice,
  formatDateWithShortMonth,
} from '../../../helpers/formatData';
import useBreakPoint from '../../../hooks/useBreakPoint';
import { selectOrders } from '../../../store/orders/ordersSlice';
import { OrderButton } from '../OrderButton/OrderButton';
import { PayButton } from '../PayButton/PayButton';

import { OrderDeliveryDate } from './OrderDeliveryDate/OrderDeliveryDate';

import 'slick-carousel/slick/slick.css';
import 'slick-carousel/slick/slick-theme.css';

import styles from './OrderCard.module.scss';

const SlickArrowLeft = ({ currentSlide, slideCount, ...props }) => (
  <ArrowBackIosNewIcon
    {...props}
    className={classNames(
      'slick-prev',
      'slick-arrow',
      currentSlide === slideCount - 1 ? ' slick-disabled' : ''
    )}
    aria-hidden="true"
    aria-disabled={currentSlide === slideCount - 1 ? true : false}
    type="button"
  />
);

const SlickArrowRight = ({ currentSlide, slideCount, ...props }) => (
  <ArrowForwardIosIcon
    {...props}
    className={classNames(
      'slick-next',
      'slick-arrow',
      currentSlide === slideCount - 1 ? ' slick-disabled' : ''
    )}
    aria-hidden="true"
    aria-disabled={currentSlide === slideCount - 1 ? true : false}
    type="button"
  />
);

SlickArrowRight.propTypes = {
  currentSlide: PropTypes.any,
  slideCount: PropTypes.any,
};

SlickArrowLeft.propTypes = {
  currentSlide: PropTypes.any,
  slideCount: PropTypes.any,
};

export function orderSlider(slides) {
  var settings = {
    dots: false,
    infinite: false,
    arrows: true,
    slidesToShow: 3,
    slidesToScroll: 1,
    nextArrow: <SlickArrowRight />,
    prevArrow: <SlickArrowLeft />,
    responsive: [
      {
        breakpoint: BREAK_POINT.LAPTOP + 200,
        settings: {
          slidesToShow: 2,
        },
      },
      {
        breakpoint: BREAK_POINT.LAPTOP - 100,
        settings: {
          slidesToShow: 1,
        },
      },
    ],
  };

  return <Slider {...settings}>{slides}</Slider>;
}

export const OrderCard = ({ orderId: id }) => {
  const orders = useSelector(selectOrders);
  const isLessThenBreakPoint = useBreakPoint(BREAK_POINT.MD);

  const [expanded, setExpanded] = useState(false);
  const [orderProducts, setOrderProducts] = useState([]);
  const index = orders.orders.findIndex((ord) => ord.id === id);

  const {
    createdAt: creationDate,
    totalDiscountPrice: price,
    deliveredAt: deliveryDate,
    deliveryAddress,
    status,
  } = orders.orders[index];

  const deliveryAddressString = `${deliveryAddress.title} ${deliveryAddress.name} ${deliveryAddress.surname} ${deliveryAddress.country.name}, ${deliveryAddress.city}, ${deliveryAddress.street}, ${deliveryAddress.building}/${deliveryAddress.flat}`;

  const lessThanDelivered = status < orderState.DELIVERED;

  const getProductsForImages = async (imagesIds, abortController) => {
    try {
      const response = await productsApi.getProductsByIds(imagesIds, {
        signal: abortController.signal,
      });

      setOrderProducts(response.data);
    } catch (err) {
      if (!abortController.signal.aborted) {
        toast.error(notificationError);
      }
    }
  };

  useEffect(() => {
    const abortController = new AbortController();
    const imagesIds = orders.orders
      .find((order) => order.id === id)
      .products.map((item) => item.originalProductId);

    getProductsForImages(imagesIds, abortController);

    return () => {
      setOrderProducts([]);
      abortController.abort();
    };
  }, [orders, id]);

  const handleExpandClick = () => setExpanded(!expanded);

  const sliderElements = orderProducts.map((product) => {
    return (
      <Link to={`${pathNames.PRODUCTS}/${product.id}`} key={product.id}>
        <CardMedia
          component="img"
          image={product.images.length ? product.images[0] : noImg}
          alt={product.name}
          className={'none'}
        />
      </Link>
    );
  });

  return (
    <Card className={styles.card}>
      <CardActions className={styles.collapseButton}>
        <Box className={styles.outerInformation}>
          <Box className={styles.dateInformation}>
            <Typography className={styles.orderDate}>
              Ordered on &nbsp;
              {isLessThenBreakPoint
                ? formatDateWithShortMonth(creationDate)
                : formatDateWithFullMonth(creationDate)}
            </Typography>
            <Typography fontSize={isLessThenBreakPoint ? '10px' : '12px'}>
              Order #{id + 1}
            </Typography>
          </Box>
          <Box>
            <Typography
              className={classNames(
                styles.deliveryMark,
                styles[`deliveryMark${status}`]
              )}
            >
              {!isLessThenBreakPoint ? orderStatus[status].deliveryStatus : ''}
            </Typography>
          </Box>
        </Box>
        <Box className={styles.priceInformation}>
          <Typography className={styles.orderPrice}>
            {formatPrice(price)} &thinsp;
          </Typography>
          <ExpandMore
            className={styles.showMoreButton}
            expand={expanded.toString()}
            onClick={handleExpandClick}
            aria-expanded={expanded}
            aria-label="show more"
          >
            <ExpandMoreIcon />
          </ExpandMore>
        </Box>
      </CardActions>
      <Collapse in={expanded} timeout="auto" unmountOnExit>
        <CardContent className={styles.cardContent}>
          <Box className={styles.description}>
            <Box className={styles.deliveryInforamtion}>
              <Typography className={styles.deliveryDestination}>
                Delivery type: &nbsp;
                <span className={styles.deliveryValue}>
                  Courier service delivery
                </span>
              </Typography>
              <Typography className={styles.deliveryDestination}>
                Delivery address: &nbsp;
                <span className={styles.deliveryValue}>
                  {deliveryAddressString}
                </span>
              </Typography>
            </Box>
            <Box className={styles.sliderContainer}>
              {orderSlider(sliderElements)}
            </Box>
          </Box>
          <Box className={styles.buttons}>
            {lessThanDelivered ? (
              <OrderButton
                className={styles.actionButton}
                orderIndex={index}
                status={status}
              />
            ) : (
              <OrderDeliveryDate status={status} deliveryDate={deliveryDate} />
            )}
            {status === orderState.WAITING_FOR_PAYMENT && (
              <PayButton orderIndex={id} />
            )}
          </Box>
        </CardContent>
      </Collapse>
    </Card>
  );
};

OrderCard.propTypes = {
  orderId: PropTypes.number,
};
