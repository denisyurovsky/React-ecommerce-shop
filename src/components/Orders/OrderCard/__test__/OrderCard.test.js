import { screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import React from 'react';

import { orderState } from '../../../../constants/orderStatus';
import { ordersDto } from '../../../../test-utils/dto/ordersDto';
import testProducts from '../../../../test-utils/dto/productsDto';
import renderWith from '../../../../test-utils/renderWith';
import { OrderCard } from '../OrderCard';

const initialUser = {
  user: {
    id: 1,
    orders: ordersDto,
  },
};

let serverUser = {
  id: 1,
  orders: ordersDto,
  isLoading: false,
  errorOccurred: false,
};

const handlersFulfilled = [
  rest.post('/confirm-order', (req, res, ctx) => {
    const { orderId } = req.body;

    return res(
      ctx.json({
        orderStatus: orderState.DELIVERED,
        orderId,
        deliveredAt: ordersDto[0].deliveredAt,
      }),
      ctx.status(200)
    );
  }),
  rest.post('/cancel-order', (req, res, ctx) => {
    const { orderId } = req.body;

    return res(
      ctx.json({ orderId, orderStatus: orderState.CANCELLED }),
      ctx.status(200)
    );
  }),
  rest.get('/orders', (req, res, ctx) => {
    const { userId } = req.body;

    const response = [];

    ordersDto.forEach((order) => {
      if (order.userId === userId) {
        response.push(order);
      }
    });

    return res(ctx.json(response), ctx.status(200));
  }),
  rest.get('/users/1', (req, res, ctx) => {
    return res(ctx.json(serverUser), ctx.status(200));
  }),
  rest.get('/users/:userId', (req, res, ctx) => {
    return res(ctx.json(serverUser), ctx.status(200));
  }),
  rest.get('/products/', (req, res, ctx) => {
    const response = [];

    if (req.url.searchParams) {
      req.url.searchParams.forEach((item) => {
        response.push(testProducts[item]);
      });
    }

    return res(ctx.json(response), ctx.status(200));
  }),
];

const serverFulfilled = setupServer(...handlersFulfilled);

describe('OrderCard component', () => {
  describe('snapshots for different order statuses', () => {
    beforeEach(() => {
      serverFulfilled.listen();
    });
    afterEach(() => {
      serverFulfilled.close();
    });

    it('Renders a valid orderCard snapshot with status equal 1', () => {
      const { asFragment } = renderWith(
        <OrderCard orderId={ordersDto[0].id} />,
        {
          orders: {
            orders: ordersDto,
          },
          user: initialUser,
        }
      );

      expect(asFragment()).toMatchSnapshot();
    });

    it('Renders a valid orderCard snapshot with status equal 2', () => {
      const { asFragment } = renderWith(
        <OrderCard orderId={ordersDto[1].id} />,
        {
          orders: {
            orders: ordersDto,
          },
          user: initialUser,
        }
      );

      expect(asFragment()).toMatchSnapshot();
    });

    it('Renders a valid orderCard snapshot with status equal 3', () => {
      const { asFragment } = renderWith(
        <OrderCard orderId={ordersDto[2].id} />,
        {
          orders: {
            orders: ordersDto,
          },
          user: initialUser,
        }
      );

      expect(asFragment()).toMatchSnapshot();
    });
    it('Renders a valid orderCard snapshot with status equal 4', () => {
      const { asFragment } = renderWith(
        <OrderCard orderId={ordersDto[3].id} />,
        {
          orders: {
            orders: ordersDto,
          },
          user: initialUser,
        }
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('reacts on cancel click correct', () => {
    beforeEach(() => {
      serverFulfilled.listen();
    });

    afterEach(() => {
      serverFulfilled.close();
    });
    it('reacts on click with status  waiting for payment', async () => {
      renderWith(<OrderCard orderId={ordersDto[0].id} />, {
        orders: {
          orders: ordersDto,
        },
        user: initialUser,
      });

      expect(screen.getByText('Waiting for payment')).toBeInTheDocument();

      const expandButton = screen.getByTestId('ExpandMoreIcon');

      userEvent.click(expandButton);

      const cancelButtons = await screen.findAllByText('Cancel order');

      userEvent.click(cancelButtons[0]);

      const cancelledMark = await screen.findByText('Cancelled');

      expect(cancelledMark).toBeInTheDocument();
    });
  });
});
