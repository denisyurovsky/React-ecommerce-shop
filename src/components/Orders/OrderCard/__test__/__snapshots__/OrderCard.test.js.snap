// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderCard component snapshots for different order statuses Renders a valid orderCard snapshot with status equal 1 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root card css-bhp9pd-MuiPaper-root-MuiCard-root"
    data-cy="orderCard"
  >
    <div
      class="MuiCardActions-root MuiCardActions-spacing collapseButton css-1t6e9jv-MuiCardActions-root"
    >
      <div
        class="outerInformation MuiBox-root css-0"
      >
        <div
          class="dateInformation MuiBox-root css-0"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 orderDate css-ahj2mt-MuiTypography-root"
          >
            Ordered on  19 April 2021
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-17893no-MuiTypography-root"
          >
            Order #2
          </p>
        </div>
        <div
          class="MuiBox-root css-0"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 deliveryMark deliveryMark1 css-ahj2mt-MuiTypography-root"
            data-cy="orderCardStatus"
          >
            Waiting for payment
          </p>
        </div>
      </div>
      <div
        class="priceInformation MuiBox-root css-0"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 orderPrice css-ahj2mt-MuiTypography-root"
        >
          10 000 $  
        </p>
        <svg
          aria-expanded="false"
          aria-hidden="true"
          aria-label="show more"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium showMoreButton css-i4bv87-MuiSvgIcon-root"
          data-testid="ExpandMoreIcon"
          expand="false"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    class="Toastify"
  />
</DocumentFragment>
`;

exports[`OrderCard component snapshots for different order statuses Renders a valid orderCard snapshot with status equal 2 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root card css-bhp9pd-MuiPaper-root-MuiCard-root"
    data-cy="orderCard"
  >
    <div
      class="MuiCardActions-root MuiCardActions-spacing collapseButton css-1t6e9jv-MuiCardActions-root"
    >
      <div
        class="outerInformation MuiBox-root css-0"
      >
        <div
          class="dateInformation MuiBox-root css-0"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 orderDate css-ahj2mt-MuiTypography-root"
          >
            Ordered on  19 January 2021
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-17893no-MuiTypography-root"
          >
            Order #3
          </p>
        </div>
        <div
          class="MuiBox-root css-0"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 deliveryMark deliveryMark2 css-ahj2mt-MuiTypography-root"
            data-cy="orderCardStatus"
          >
            Waiting for delivery
          </p>
        </div>
      </div>
      <div
        class="priceInformation MuiBox-root css-0"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 orderPrice css-ahj2mt-MuiTypography-root"
        >
          714 $  
        </p>
        <svg
          aria-expanded="false"
          aria-hidden="true"
          aria-label="show more"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium showMoreButton css-i4bv87-MuiSvgIcon-root"
          data-testid="ExpandMoreIcon"
          expand="false"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    class="Toastify"
  />
</DocumentFragment>
`;

exports[`OrderCard component snapshots for different order statuses Renders a valid orderCard snapshot with status equal 3 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root card css-bhp9pd-MuiPaper-root-MuiCard-root"
    data-cy="orderCard"
  >
    <div
      class="MuiCardActions-root MuiCardActions-spacing collapseButton css-1t6e9jv-MuiCardActions-root"
    >
      <div
        class="outerInformation MuiBox-root css-0"
      >
        <div
          class="dateInformation MuiBox-root css-0"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 orderDate css-ahj2mt-MuiTypography-root"
          >
            Ordered on  19 March 2022
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-17893no-MuiTypography-root"
          >
            Order #4
          </p>
        </div>
        <div
          class="MuiBox-root css-0"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 deliveryMark deliveryMark3 css-ahj2mt-MuiTypography-root"
            data-cy="orderCardStatus"
          >
            Delivered
          </p>
        </div>
      </div>
      <div
        class="priceInformation MuiBox-root css-0"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 orderPrice css-ahj2mt-MuiTypography-root"
        >
          539 $  
        </p>
        <svg
          aria-expanded="false"
          aria-hidden="true"
          aria-label="show more"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium showMoreButton css-i4bv87-MuiSvgIcon-root"
          data-testid="ExpandMoreIcon"
          expand="false"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    class="Toastify"
  />
</DocumentFragment>
`;

exports[`OrderCard component snapshots for different order statuses Renders a valid orderCard snapshot with status equal 4 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root card css-bhp9pd-MuiPaper-root-MuiCard-root"
    data-cy="orderCard"
  >
    <div
      class="MuiCardActions-root MuiCardActions-spacing collapseButton css-1t6e9jv-MuiCardActions-root"
    >
      <div
        class="outerInformation MuiBox-root css-0"
      >
        <div
          class="dateInformation MuiBox-root css-0"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 orderDate css-ahj2mt-MuiTypography-root"
          >
            Ordered on  19 September 2021
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-17893no-MuiTypography-root"
          >
            Order #5
          </p>
        </div>
        <div
          class="MuiBox-root css-0"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 deliveryMark deliveryMark4 css-ahj2mt-MuiTypography-root"
            data-cy="orderCardStatus"
          >
            Cancelled
          </p>
        </div>
      </div>
      <div
        class="priceInformation MuiBox-root css-0"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 orderPrice css-ahj2mt-MuiTypography-root"
        >
          274,50 $  
        </p>
        <svg
          aria-expanded="false"
          aria-hidden="true"
          aria-label="show more"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium showMoreButton css-i4bv87-MuiSvgIcon-root"
          data-testid="ExpandMoreIcon"
          expand="false"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    class="Toastify"
  />
</DocumentFragment>
`;
